### PR TITLE
Outlook spam reporting sample

### PR DESCRIPTION
**Pull request by elegault**
|        Q        |                    A                    |
| --------------- | --------------------------------------- |
| Bug fix?        | no                                |
| New feature?    | no                             |
| New sample?     |  yes                               |
| Related issues? | no |

## What's in this Pull Request?

> Please describe the changes in this PR. Sample description or details around bugs which are being fixed.
> 
Code sample to supplement the [Implement an integrated spam-reporting add-in (preview)](https://learn.microsoft.com/en-us/office/dev/add-ins/outlook/spam-reporting) article.

## Guidance

> *Please update this PR information accordingly. We'll use this as part of our release notes in monthly communications.*
> 
> *Please target your PR to `dev` branch. Released documents are in `main` branch.*

### CODE REVIEW

**Comment by elegault** [comment]
Changed target branch to main, as per @davidchesnut. I swear the [contributing guidelines](https://github.com/OfficeDev/Office-Add-in-samples/blob/77feb5f033d4bf387f02592d2150a2a8c56ad1d7/CONTRIBUTING.md) said "dev", but it is clearly correct so not sure what happened or if I saw a different version somewhere.

File: Samples/outlook-spam-reporting/manifest.xml [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+      <DesktopSettings>
+        <SourceLocation DefaultValue="https://officedev.github.io/Office-Add-in-samples/Samples/ericl-outlook-spam-reporting-sample/commands/spamreporting.html"/>
+        <RequestedHeight>250</RequestedHeight>
+      </DesktopSettings>
+    </Form>
+  </FormSettings>
+
+  <Permissions>ReadItem</Permissions>
+  <Rule xsi:type="RuleCollection" Mode="Or">
+    <Rule xsi:type="ItemIs" ItemType="Message" FormType="Read" />
```
Comment (samantharamon) [inline Samples/outlook-spam-reporting/manifest.xml:49]:
```suggestion
    <Rule xsi:type="ItemIs" ItemType="Message" FormType="Read"/>
```

File: Samples/outlook-spam-reporting/manifest.xml [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+      <Resources>
+        <bt:Images>
+          <bt:Image id="Icon.16x16" DefaultValue="https://officedev.github.io/Office-Add-in-samples/Samples/ericl-outlook-spam-reporting-sample/assets/icon-16.png"/>
+          <bt:Image id="Icon.32x32" DefaultValue="https://officedev.github.io/Office-Add-in-samples/Samples/ericl-outlook-spam-reporting-sample/assets/icon-32.png"/>
+          <bt:Image id="Icon.80x80" DefaultValue="https://officedev.github.io/Office-Add-in-samples/Samples/ericl-outlook-spam-reporting-sample/assets/icon-80.png"/>
+        </bt:Images>
+        <bt:Urls>
+          <bt:Url id="WebViewRuntime.Url" DefaultValue="https://officedev.github.io/Office-Add-in-samples/Samples/ericl-outlook-spam-reporting-sample/commands/spamreporting.html"/>
+          <bt:Url id="JSRuntime.Url" DefaultValue="https://officedev.github.io/Office-Add-in-samples/Samples/ericl-outlook-spam-reporting-sample/commands/spamreporting.js"/>
+          <bt:Url id="MoreInfo.Url" DefaultValue="https://learn.microsoft.com/en-us/office/dev/add-ins/outlook/spam-reporting"/>
```
Comment (samantharamon) [inline Samples/outlook-spam-reporting/manifest.xml:121]:
```suggestion
          <bt:Url id="MoreInfo.Url" DefaultValue="https://learn.microsoft.com/office/dev/add-ins/outlook/spam-reporting"/>
```

File: Samples/outlook-spam-reporting/src/commands/spamreporting.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
+﻿/*
```
Comment (samantharamon) [inline Samples/outlook-spam-reporting/src/commands/spamreporting.js:1]: Consider moving the spamreporting.html and spamreporting.js files directly to the src folder. This removes the additional folder level ("commands"). If you make this change, you may need to update the file references in the manifest.

**Review by samantharamon — CHANGES_REQUESTED** [review]
@elegault thank you for all your contributions to the Outlook add-ins space! I added some suggestions for your review.

**Review by elegault — COMMENTED** [review]
Hi @samantharamon - I made the changes as requested.

File: Samples/outlook-spam-reporting/src/commands/spamreporting.js (hunk 3 above)
Comment (elegault) [reply Samples/outlook-spam-reporting/src/commands/spamreporting.js]: Done

File: Samples/outlook-spam-reporting/README.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+> [!IMPORTANT]
+> To test the `getAsFileAsync` method while it's still in preview in Outlook on Windows, you must configure your computer's registry.
+>
+> Outlook on Windows includes a local copy of the production and beta versions of Office.js instead of loading from the content delivery network (CDN). By default, the local production copy of the API is referenced. To reference the local beta copy of the API, you must configure your computer's registry as follows:
+>
+> 1. In the registry, navigate to `HKEY_CURRENT_USER\SOFTWARE\Microsoft\Office\16.0\Outlook\Options\WebExt\Developer`. If the key doesn't exist, create it.
+> 1. Create an entry named `EnableBetaAPIsInJavaScript` and set its value to `1`.
+>
+>    :::image type="content" source="../images/outlook-beta-registry-key.png" alt-text="The EnableBetaAPIsInJavaScript registry value is set to 1.":::
```
Comment (samantharamon) [inline Samples/outlook-spam-reporting/README.md:59]:
```suggestion
>    ![The EnableBetaAPIsInJavaScript registry value is set to 1.](https://learn.microsoft.com/office/dev/add-ins/images/outlook-beta-registry-key.png)
```

File: Samples/outlook-spam-reporting/README.md [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+- Is easily discoverable in the Outlook client ribbon.
+- Provides the user with a processing dialog to report an email.
+- Facilitates saving a copy of the reported email to a file to submit it to your backend system for further processing.
+
+To learn about key components of this sample, see [Implement an integrated spam-reporting add-in (preview)](https://learn.microsoft.com/office/dev/add-ins/outlook/spam-reporting).
+
+> [!IMPORTANT]
+
+> The integrated spam-reporting feature is currently in preview in Outlook on Windows. Features in preview shouldn't be used in production add-ins. We invite you to try out this feature in test or development environments and welcome feedback on your experience through [GitHub](https://github.com/OfficeDev/office-js/issues/new/choose).
```
Comment (samantharamon) [inline Samples/outlook-spam-reporting/README.md:37]:
```suggestion
> [!IMPORTANT]
> The integrated spam-reporting feature is currently in preview in Outlook on Windows. Features in preview shouldn't be used in production add-ins. We invite you to try out this feature in test or development environments and welcome feedback on your experience through [GitHub](https://github.com/OfficeDev/office-js/issues/new/choose).
```

File: Samples/outlook-spam-reporting/README.md [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+
+## Applies to
+
+Outlook on Windows starting in Version 2307 (Build 16626.10000).
+    > [!NOTE]
+    > If you don't have a Microsoft 365 subscription, you might qualify for a free developer subscription that's renewable for 90 days and comes configured with sample data. For details, see the [Microsoft 365 Developer Program FAQ](https://learn.microsoft.com/office/developer-program/microsoft-365-developer-program-faq#who-qualifies-for-a-microsoft-365-e5-developer-subscription-).
```
Comment (samantharamon) [inline Samples/outlook-spam-reporting/README.md:43]:
```suggestion

> [!NOTE]
> If you don't have a Microsoft 365 subscription, you might qualify for a free developer subscription that's renewable for 90 days and comes configured with sample data. For details, see the [Microsoft 365 Developer Program FAQ](https://learn.microsoft.com/office/developer-program/microsoft-365-developer-program-faq#who-qualifies-for-a-microsoft-365-e5-developer-subscription-).
```

File: Samples/outlook-spam-reporting/README.md [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+  - Add-ins
+  createdDate: 02/12/2024 10:00:00 AM
+description: "Learn how to create an integrated spam-reporting add-in in Outlook."
+---
+
+# Report spam or phishing emails in Outlook (preview)
+
+**Applies to**: Outlook on Windows (preview)
+
+[A sample spam-reporting dialog.](/assets/readme/outlook-spam-processing-dialog.png)
```
Comment (samantharamon) [inline Samples/outlook-spam-reporting/README.md:23]:
```suggestion
![A sample spam-reporting dialog.](/assets/readme/outlook-spam-processing-dialog.png)
```

File: Samples/outlook-spam-reporting/README.md [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+
+Outlook on Windows starting in Version 2307 (Build 16626.10000).
+    > [!NOTE]
+    > If you don't have a Microsoft 365 subscription, you might qualify for a free developer subscription that's renewable for 90 days and comes configured with sample data. For details, see the [Microsoft 365 Developer Program FAQ](https://learn.microsoft.com/office/developer-program/microsoft-365-developer-program-faq#who-qualifies-for-a-microsoft-365-e5-developer-subscription-).
+
+## Prerequisites
+- Microsoft 365 subscription
+
+    > [!TIP]
+    > If you're unable to choose a channel in your Outlook client, see [Let users choose which Microsoft 365 Insider channel to install on Windows devices](https://learn.microsoft.com/deployoffice/insider/deploy/user-choice).
```
Comment (samantharamon) [inline Samples/outlook-spam-reporting/README.md:49]:
```suggestion
> [!TIP]
> If you're unable to choose a channel in your Outlook client, see [Let users choose which Microsoft 365 Insider channel to install on Windows devices](https://learn.microsoft.com/deployoffice/insider/deploy/user-choice).
```

File: Samples/outlook-spam-reporting/README.md [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+1. Now that your localhost web server is running, you can sideload the **manifest-localhost.xml** file provided in the sample folder. To sideload the manifest, follow the manual instructions in [Sideload Outlook add-ins for testing](https://learn.microsoft.com/office/dev/add-ins/outlook/sideload-outlook-add-ins-for-testing?tabs=windows-web#sideload-manually).
+1. Follow the steps in [Try it out](#try-it-out) to test the sample.
+
+## Try it out
+
+Once the add-in is loaded in Outlook, use the following steps to try out its functionality.
+
+1. Choose a message from your inbox, then select the add-in's button from the ribbon.
+
+    [The spam-reporting add-in button is selected from the ribbon.](/assets//readme/outlook-spam-ribbon-button.png)
```
Comment (samantharamon) [inline Samples/outlook-spam-reporting/README.md:115]:
```suggestion
    ![The spam-reporting add-in button is selected from the ribbon.](/assets//readme/outlook-spam-ribbon-button.png)
```

File: Samples/outlook-spam-reporting/README.md [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+1. In the preprocessing dialog, choose a reason for reporting the message and add information about the message, if configured. Then, select **Report**.
+1. (Optional) In the post-processing dialog, select **OK**.
+
+    [The post-processing dialog of the sample spam-reporting add-in.](/assets//readme/outlook-spam-post-processing-dialog.png)
```
Comment (samantharamon) [inline Samples/outlook-spam-reporting/README.md:119]:
```suggestion
    ![The post-processing dialog of the sample spam-reporting add-in.](/assets//readme/outlook-spam-post-processing-dialog.png)
```

File: Samples/outlook-spam-reporting/README.md [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+  - Add-ins
+  createdDate: 03/26/2024 10:00:00 AM
+description: "Learn how to create an integrated spam-reporting add-in in Outlook."
+---
+
+# Report spam or phishing emails in Outlook (preview)
+
+**Applies to**: Outlook on Windows (preview)
+
+![A sample spam-reporting dialog.](/assets/readme/outlook-spam-processing-dialog.png)
```
Comment (samantharamon) [inline Samples/outlook-spam-reporting/README.md:23]:
```suggestion
![A sample spam-reporting dialog.](./assets/readme/outlook-spam-processing-dialog.png)
```

File: Samples/outlook-spam-reporting/README.md [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+1. Now that your localhost web server is running, you can sideload the **manifest-localhost.xml** file provided in the sample folder. To sideload the manifest, follow the manual instructions in [Sideload Outlook add-ins for testing](https://learn.microsoft.com/office/dev/add-ins/outlook/sideload-outlook-add-ins-for-testing?tabs=windows-web#sideload-manually).
+1. Follow the steps in [Try it out](#try-it-out) to test the sample.
+
+## Try it out
+
+Once the add-in is loaded in Outlook, use the following steps to try out its functionality.
+
+1. Choose a message from your inbox, then select the add-in's button from the ribbon.
+
+    ![The spam-reporting add-in button is selected from the ribbon.](/assets//readme/outlook-spam-ribbon-button.png)
```
Comment (samantharamon) [inline Samples/outlook-spam-reporting/README.md:115]:
```suggestion
    ![The spam-reporting add-in button is selected from the ribbon.](./assets//readme/outlook-spam-ribbon-button.png)
```

File: Samples/outlook-spam-reporting/README.md [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+1. In the preprocessing dialog, choose a reason for reporting the message and add information about the message, if configured. Then, select **Report**.
+1. (Optional) In the post-processing dialog, select **OK**.
+
+    ![The post-processing dialog of the sample spam-reporting add-in.](/assets//readme/outlook-spam-post-processing-dialog.png)
```
Comment (samantharamon) [inline Samples/outlook-spam-reporting/README.md:119]:
```suggestion
    ![The post-processing dialog of the sample spam-reporting add-in.](./assets//readme/outlook-spam-post-processing-dialog.png)
```

**Review by samantharamon — APPROVED** [review]
Thanks for making the changes, @elegault. I made additional formatting changes to fix the alerts and images. Everything else looks good to me, so I'll go ahead and publish the sample. Thank you again for your contributions to Outlook extensibility!

**Pull request merged**